test/Abstractions.Tests/TaskOptionsTests.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.Tests;
+
+public class TaskOptionsTests
+{
+    [Fact]
+    public void Empty_Ctors_Okay()
+    {
+        TaskOptions options = new();
+        options.Retry.Should().BeNull();
+        options.Tags.Should().BeNull();
+
+        SubOrchestrationOptions subOptions = new();
+        subOptions.Retry.Should().BeNull();
+        subOptions.Tags.Should().BeNull();
+        subOptions.InstanceId.Should().BeNull();
+
+        StartOrchestrationOptions startOptions = new();
+        startOptions.Version.Should().BeNull();
+        startOptions.InstanceId.Should().BeNull();
+        startOptions.StartAt.Should().BeNull();
+        startOptions.Tags.Should().BeEmpty();
```

Comment:
Weird that tags defaults to empty here but defaults to null in the *Options classes. Oh well - not something to deal with in this PR.